### PR TITLE
topology: fix buffer size calculation

### PR DESCRIPTION
**Pull request by juimonen**
The buffer frame size calculation has obvious issues by
doing division instead of multiplication. On the other
hand we can't do decimal calculations in M4. So fix this
by introducing new macro for buffer frame size where we
divide samplerate and schedule_period first by 1000 before
multiplication. This should work for scheduling times
not smaller than 1000us.

Signed-off-by: Jaska Uimonen <jaska.uimonen@intel.com>

### CODE REVIEW

**Comment by juimonen** [comment]
should fix https://github.com/thesofproject/sof/issues/1727

File: tools/topology/sof/pipe-demux-playback.m4 [hunk 1]
```diff
@@ -46,7 +46,7 @@ W_MUXDEMUX(0, 1, PIPELINE_FORMAT, 2, 2, LIST(`         ', "DEMUX"))
 
 # Low Latency Buffers
 W_BUFFER(0, COMP_BUFFER_SIZE(2,
-	COMP_SAMPLE_SIZE(PIPELINE_FORMAT), PIPELINE_CHANNELS, eval(PCM_MAX_RATE / SCHEDULE_PERIOD)),
+	COMP_SAMPLE_SIZE(PIPELINE_FORMAT), PIPELINE_CHANNELS, COMP_BUFFER_FRAMES(PCM_MAX_RATE, SCHEDULE_PERIOD)),
```
Comment (ranj063) [inline tools/topology/sof/pipe-demux-playback.m4]: @juimonen just wondering on line 52 of this file for the W_BUFFER(1...), there's a val(). is this a typo?

Comment (juimonen) [reply tools/topology/sof/pipe-demux-playback.m4]: yes, thanks for pointing out. updated now.

**Comment by jajanusz** [comment]
@juimonen Please resolve conflicts

**Comment by juimonen** [comment]
@jajanusz pushed new version

File: tools/topology/m4/buffer.m4 [hunk 2]
```diff
@@ -29,4 +29,6 @@ define(`W_BUFFER',
 dnl COMP_BUFFER_SIZE( num_periods, sample_size, channels, fmames)
 define(`COMP_BUFFER_SIZE', `eval(`$1 * $2 * $3 * $4')')
 
+define(`COMP_BUFFER_FRAMES', `eval(`($1 / 1000) * ($2 / 1000)')')
```
Comment (keyonjie) [inline tools/topology/m4/buffer.m4]: the naming 'COMP_BUFFER_FRAMES' is somewhat hard for understanding, how about 'COMP_PERIOD_FRAMES'?

Comment (juimonen) [reply tools/topology/m4/buffer.m4]: @keyonjie sure I can do it if it is more clear... just a little bit error prone search-replace-regexp for me... if not absolutely needed, I wouldn't do it :)

Comment (keyonjie) [reply tools/topology/m4/buffer.m4]: BTW, how about the case scheduling times smaller than 1000us? And add some comment about the macro also.
Can we change it to:
```
dnl COMP_PERIOD_SIZE( sample_rate, period_us)
define(`COMP_PERIOD_FRAMES', `eval(`($1 * $2) / 1000000')')
```

Comment (juimonen) [reply tools/topology/m4/buffer.m4]: @keyonjie yes that looks better, I can change for sure.

**Comment by juimonen** [comment]
@keyon updated and changed the commit message a little bit.

**Pull request merged**